wallet/lib/stores/wallet_store_imp.dart

```diff
@@ -266,19 +270,68 @@ class WalletsStoreImp implements WalletsStore {
     return response;
   }
 
+  LocalTransactionModel createInitialLocalTransactionModel({
+    required TransactionTypeEnum transactionTypeEnum,
+    required String transactionData,
+    required String transactionCurrency,
+    required String transactionPrice,
+    required String transactionDescription,
+  }) {
+    final LocalTransactionModel txManager = LocalTransactionModel(
+        transactionType: transactionTypeEnum.name,
+        transactionData: transactionData,
+        transactionCurrency: transactionCurrency,
+        transactionPrice: transactionPrice,
+        transactionDescription: transactionDescription,
+        dateTime: DateTime.now().millisecondsSinceEpoch,
+        status: TransactionStatus.Undefined.name);
+    return txManager;
+  }
+
   @override
   Future<SdkIpcResponse> createTrade(Map json) async {
     final msgObj = pylons.MsgCreateTrade.create()..mergeFromProto3Json(json);
     msgObj.creator = wallets.value.last.publicAddress;
     return _signAndBroadcast(msgObj);
   }
 
+  Future<void> saveTransactionRecord({required TransactionStatus transactionStatus, required LocalTransactionModel txLocalModel}) async {
+    final txLocalModelWithStatus = LocalTransactionModel.fromStatus(status: transactionStatus, transactionModel: txLocalModel);
+    repository.saveLocalTransaction(txLocalModelWithStatus);
+  }
+
   @override
-  Future<SdkIpcResponse> executeRecipe(Map json) async {
-    final msgObj = pylons.MsgExecuteRecipe.create()..mergeFromProto3Json(json);
-    msgObj.creator = wallets.value.last.publicAddress;
-    final response = await _signAndBroadcast(msgObj);
-    return response;
+  Future<Either<Failure, SdkIpcResponse>> executeRecipe(Map json) async {
+    final networkInfo = GetIt.I.get<NetworkInfo>();
+
+    final LocalTransactionModel localTransactionModel = createInitialLocalTransactionModel(
+      transactionTypeEnum: TransactionTypeEnum.BuyNFT,
+      transactionData: jsonEncode(json),
+      transactionDescription: "${'bought_nft'.tr()}  ${json[kNftName] ?? ""}",
+      transactionCurrency: "${json[kNftCurrency] ?? ""}",
+      transactionPrice: "${json[kNftPrice] ?? ""}",
+    );
+
+    if (!await networkInfo.isConnected) {
+      await saveTransactionRecord(transactionStatus: TransactionStatus.Failed, txLocalModel: localTransactionModel);
+      return Left(NoInternetFailure("no_internet".tr()));
+    }
+    try {
+      json.remove(kNftName);
```

Comment:
Why we are removing them?

Reply:
Because this same json will be sent to proto file and they only take the existing parameters. We are just passing them to get price details for failure screens.